test/handlers/FranchiserFactoryHandler.sol

```diff
@@ -2,63 +2,197 @@
+
+    // Ghost variable address to receive of the total amount of funds recalled from franchisers
+    address public targetAddressForRecalledFunds = makeAddr("targetAddressForRecalledFunds");
 
     constructor(FranchiserFactory _factory) {
         factory = _factory;
-        franchiser = new Franchiser(IVotingToken(address(factory.votingToken())));
+        votingToken = VotingTokenConcrete(address(factory.votingToken()));
+        franchiser = new Franchiser(IVotingToken(address(votingToken)));
     }
 
-    function _validActorAddress(address _address) internal view returns (bool valid) {
-        valid =
-            (_address != address(0)) && (_address != address(factory.votingToken()) && (_address != address(factory)));
+    modifier countCall(bytes32 key) {
+        calls[key]++;
+        _;
     }
 
-    function _boundAmount(uint256 _amount) internal pure returns (uint256) {
-        return bound(_amount, 0, 100_000_000e18);
+    function sumFundedFranchisersBalances() public view returns (uint256 sum) {
+        for (uint256 i = 0; i < fundedFranchisers.length(); i++) {
+            sum += factory.votingToken().balanceOf(fundedFranchisers.at(i));
+        }
     }
 
-    function _removeFranchiser(uint256 _index) public {
-        require(_index < fundedFranchisers.length, "Index out of bounds");
+    function _validActorAddress(address _address) internal view returns (bool valid) {
+        valid = (_address != address(0))
+            && (
+                _address != address(votingToken) && (_address != address(factory))
+                    && (!fundedFranchisers.contains(_address))
+            );
+    }
 
-        for (uint256 i = _index; i < fundedFranchisers.length - 1; i++) {
-            fundedFranchisers[i] = fundedFranchisers[i + 1];
-        }
-        fundedFranchisers.pop();
+    function _boundAmount(uint256 _amount) internal pure returns (uint256) {
+        return bound(_amount, 0, 100_000_000e18);
     }
 
-    function handler_fund(address _delegator, address _delegatee, uint256 _amount) external {
+    function handler_fund(address _delegator, address _delegatee, uint256 _amount) external countCall("handler_fund") {
         vm.assume(_validActorAddress(_delegator));
         _amount = _boundAmount(_amount);
-        VotingTokenConcrete votingToken = VotingTokenConcrete(address(factory.votingToken()));
         votingToken.mint(_delegator, _amount);
         vm.startPrank(_delegator);
         votingToken.approve(address(factory), _amount);
         franchiser = factory.fund(_delegatee, _amount);
         vm.stopPrank();
-        fundedFranchisers.push(franchiser);
+        fundedFranchisers.add(address(franchiser));
+    }
+
+    function handler_fundMany(address _delegator, address[] memory _delegatees, uint256 _baseAmount)
+        external
+        countCall("handler_fundMany")
+    {
+        uint256 _numberOfDelegatees = _delegatees.length;
+        _baseAmount = _bound(_baseAmount, 1, 10_000e18);
```

Comment:
Maybe bound this to an upper bound of 10 million

Reply:
Done: 9211e4a5554f20c1e9f74eb5a031d4a5a817f049